File: algorithm/5. Longest Palindromic Substring/5.py
class Solution(object):
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: str
        """
        size = len(s)
        max1=0
        start=0
        for i in xrange(size):
            j = i-1;
            k = i+1;
            while j >=0 and k < size and s[j]==s[k]:
                j -= 1
                k += 1
            j += 1
            k -= 1
            if k - j + 1 > max1:
                max1 = k - j + 1
                start = j
        for i in xrange(size):
            j = i;
            k = i+1;
            while j >=0 and k < size and s[j]==s[k]:
                j -= 1
                k += 1
            j += 1
            k -= 1
            if k - j + 1 > max1:
                max1 = k - j + 1
                start = j
        if max > 0:
            return s[start : start + max1]
        return None
        

File: algorithm/5. Longest Palindromic Substring/5.cpp
class Solution {
public:
    string longestPalindrome(string s) {
        int len = s.length();
        int max=0;
        int start = 0;
        for( int i =0 ; i < len; i++ )
        {
            int j = i - 1;
            int k = i + 1;
            while ( j >=0 && k < len && s[j]==s[k])
            {
                j--;
                k++;
            }
            j++;
            k--;
            if(k-j+1>max)
            {
                max=k-j+1;
                start=j;
            }
        }
        for( int i =0 ; i < len; i++ )
        {
            int j = i;
            int k = i + 1;
            while ( j >=0 && k < len && s[j]==s[k])
            {
                j--;
                k++;
            }
            j++;
            k--;
            if(k-j+1>max)
            {
                max=k-j+1;
                start=j;
            }
        }
        
        if (max>0)
        {
            return s.substr(start,  max);
        }
        else
            return NULL;
        
    }
};
